### DISCUSSION TITLE
.findOne returns array

### DISCUSSION THREAD

**Discussion by CMarker** [sequelize/sequelize#15515] [Help & Questions] [answered]
The documentation implies the .findOne function should return a single model object - https://sequelize.org/docs/v6/core-concepts/model-querying-finders/#findone

when I do `let auth = await db.api_auths.findOne({where: {id: 123}});` it is returning `[{id: 123}]` - an object nested in an array. this is a breaking change from a previous version but I cannot find any reference to this anywhere or any work arounds other than refactoring a large part of the code base.

**Comment by ephys** [comment, marked as answer]
If you're experiencing that with the query you provided, it's likely a bug, but we'd need a [SSCCE](https://github.com/sequelize/sequelize-sscce) to be able to replicate it and fix it

**Reply by CMarker** [reply]
Thank you for this suggestion. It wasn't the direct answer but it did give me a working sane test case to compare.

My specific problem was overriding the .query function to support old style syntax in a legacy project and not properly supporting all edge cases. I didnt realize that .findOne called query internally and my override interfered with it but that makes sense.

Thank you for such a fast response! I didn't expect to hear anything back for a couple days. 

Happy New Year!
